src/redux/books/booksSlice.js

```diff
@@ -1,39 +1,107 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 
-const initialState = [
-  {
-    id: 1,
-    title: 'Code Complete',
-    author: 'Steve McConnell',
-  },
-  {
-    id: 2,
-    title: 'Programming Pearls',
-    author: 'Jon Bentley',
-  },
-  {
-    id: 3,
-    title: 'Cacking the Coding Interview',
-    author: 'Gayle Laakmann',
-  },
-];
+const getBooksURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/wUxXlMuiwuovrvJw2spJ/books';
+
+// get books from the API
+export const getBooks = createAsyncThunk('getBooks', async (thunkAPI) => {
+  try {
+    const response = await fetch(getBooksURL);
+    return response.json();
+  } catch (error) {
+    return thunkAPI.rejectWithValue({ error: error.message });
+  }
+});
+
+// Add a book to API
+export const addBookURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/wUxXlMuiwuovrvJw2spJ/books';
+export const postBook = createAsyncThunk('books/addBook', async (book, thunkAPI) => {
+  try {
+    const response = await fetch(addBookURL, {
+      method: 'POST',
+      body: JSON.stringify(book),
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+    if (response.ok) {
+      const data = await response.json();
+      return data;
+    }
+    throw new Error('Failed to add book');
+  } catch (error) {
+    return thunkAPI.rejectWithValue('Something went wrong');
+  }
+});
+
+// Remove a book from API
+export const deleteBookURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/wUxXlMuiwuovrvJw2spJ/books';
+export const deleteBook = createAsyncThunk('books/deleteBook', async (bookId, thunkAPI) => {
+  try {
+    const response = await fetch(`${deleteBookURL}/${bookId}`, {
+      method: 'DELETE',
+    });
+
+    if (response.ok) {
+      const data = await response.json();
+      return data;
+    }
+    throw new Error('Failed to delete book');
+  } catch (error) {
+    return thunkAPI.rejectWithValue('Something went wrong');
+  }
+});
 
 const booksSlice = createSlice({
   name: 'books',
-  initialState,
+  initialState: {
+    isLoading: false,
+    data: null,
+    isError: false,
+    books: [],
+  },
   reducers: {
-    addBook: (state, action) => {
-      const newBook = { ...action.payload, id: state.length + 1 };
-      state.push(newBook);
-    },
     removeBook: (state, action) => {
-      const bookIndex = state.findIndex((book) => book.id === action.payload);
-      if (bookIndex !== -1) {
-        state.splice(bookIndex, 1);
-      }
+      const bookId = action.payload;
+      /* eslint-disable-next-line no-param-reassign */
```

Comment:
I noticed a lot of eslint disable statements in your code. I know some linter errors can be frustrating but please try to solve all of them without the use of eslint disable statements. This way you will improve yourself as a developer and know how to adopt best coding practices. Please remove all eslint disable statements and try to solve the linter errors yourself🙏

Reply:
I have removed all the `eslint-disable` statement from the code.